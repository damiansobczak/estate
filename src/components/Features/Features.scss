@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.features {
  margin: 80px 0;
  border: 1px solid transparent;
  &__desc {
    flex-direction: column;
  }
  &__title {
    color: $color-text-primary;
  }
  &__text {
    color: $color-text-secondary;
    line-height: 1.5;
    font-weight: 500;
    margin-bottom: 50px;
  }
}
