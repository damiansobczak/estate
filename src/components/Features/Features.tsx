import React from "react";
import "./Features.scss";
import FeaturesItem from "../FeaturesItem/FeaturesItem";

class Features extends React.Component<any, any> {
  render() {
    return (
      <section className="features">
        <div className="container">
          <div className="features__desc columns">
            <h4 className="features__title column is-6-tablet">Cały czas zastanawiasz się dlaczego powinieneś wybrać nasze usługi?</h4>
            <p className="features__text column is-8-tablet">Poznaj kilka faktów o naszej firmie. Mamy nadzieję że rozjaśni to Twoje wątpliwości. Pamiętaj że nasi konsultanci pozostają do Twojej dyspozycji!</p>
          </div>
          <div className="features__items columns">
            <FeaturesItem number="1" icon="key" title="Najniższe ceny w mieście" />
            <FeaturesItem number="2" icon="check-square" title="Wszystko załatwimy za Ciebie" />
            <FeaturesItem number="3" icon="gift" title="Dajemy roczną gwarancję" />
            <FeaturesItem number="4" icon="volume-2" title="Stawiamy na komunikację" />
            <FeaturesItem number="5" icon="unlock" title="Bezpieczeństwo transakcji" />
          </div>
        </div>
      </section>
    );
  }
}

export default Features;
