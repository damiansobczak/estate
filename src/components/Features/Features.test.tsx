import React from "react";
import { render } from "enzyme";
import renderer from "react-test-renderer";
import Features from "./Features";

describe("<Features />", () => {
    it("renders component correctly", () => {
        const tree = renderer.create(<Features />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("displays title correctly", () => {
        const wrapper = render(<Features />);
        expect(wrapper.find(".features__title").html()).toContain("Ca&#x142;y czas zastanawiasz si&#x119; dlaczego powiniene&#x15B; wybra&#x107; nasze us&#x142;ugi?");
    });

    it("displays text correctly", () => {
        const wrapper = render(<Features />);
        expect(wrapper.find(".features__title").html()).toContain("Ca&#x142;y czas zastanawiasz si&#x119; dlaczego powiniene&#x15B; wybra&#x107; nasze us&#x142;ugi?");
    });

    it("displays correct number of items", () => {
        const wrapper = render(<Features />);
        expect(wrapper.find(".features__items").children().length).toEqual(5);
    });
});
