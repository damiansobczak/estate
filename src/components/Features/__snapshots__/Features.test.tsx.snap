// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Features /> renders component correctly 1`] = `
<section
  className="features"
>
  <div
    className="container"
  >
    <div
      className="features__desc columns"
    >
      <h4
        className="features__title column is-6-tablet"
      >
        Cały czas zastanawiasz się dlaczego powinieneś wybrać nasze usługi?
      </h4>
      <p
        className="features__text column is-8-tablet"
      >
        Poznaj kilka faktów o naszej firmie. Mamy nadzieję że rozjaśni to Twoje wątpliwości. Pamiętaj że nasi konsultanci pozostają do Twojej dyspozycji!
      </p>
    </div>
    <div
      className="features__items columns"
    >
      <div
        className="featuresItem column"
      >
        <div
          className="featuresItem__icon"
          data-number="1"
        >
          <span
            className="icon-key"
          />
        </div>
        <div
          className="featuresItem__title"
        >
          Najniższe ceny w mieście
        </div>
      </div>
      <div
        className="featuresItem column"
      >
        <div
          className="featuresItem__icon"
          data-number="2"
        >
          <span
            className="icon-check-square"
          />
        </div>
        <div
          className="featuresItem__title"
        >
          Wszystko załatwimy za Ciebie
        </div>
      </div>
      <div
        className="featuresItem column"
      >
        <div
          className="featuresItem__icon"
          data-number="3"
        >
          <span
            className="icon-gift"
          />
        </div>
        <div
          className="featuresItem__title"
        >
          Dajemy roczną gwarancję
        </div>
      </div>
      <div
        className="featuresItem column"
      >
        <div
          className="featuresItem__icon"
          data-number="4"
        >
          <span
            className="icon-volume-2"
          />
        </div>
        <div
          className="featuresItem__title"
        >
          Stawiamy na komunikację
        </div>
      </div>
      <div
        className="featuresItem column"
      >
        <div
          className="featuresItem__icon"
          data-number="5"
        >
          <span
            className="icon-unlock"
          />
        </div>
        <div
          className="featuresItem__title"
        >
          Bezpieczeństwo transakcji
        </div>
      </div>
    </div>
  </div>
</section>
`;
